common/workunit/workunit.cpp

```diff
@@ -7986,10 +8055,20 @@ void CLocalWorkUnit::setStatistic(StatisticCreatorType creatorType, const char *
         else
             statTree->removeProp("@max");
     }
-    if (creatorType==SCTsummary && kind==StTimeElapsed && isGlobalScope(scope))
+
+    //Whenever a graph time is updated recalculate the total time spent in thor, and save it
+    if ((scopeType == SSTgraph) && (kind == StTimeElapsed))
     {
+        _loadStatistics();
+        stat_type totalTime = 0;
+        ForEachItemIn(i, statistics)
+        {
+            IConstWUStatistic & cur = statistics.item(i);
+            if ((cur.getScopeType() == SSTgraph) && (cur.getKind() == StTimeElapsed))
+                totalTime += cur.getValue();
+        }
         StringBuffer t;
-        formatTimeCollatable(t, value, false);
+        formatTimeCollatable(t, totalTime, false);
```

Comment:
@ghalliday Is each thor node going to be updateing the graph time after it updates their stats?  Won't extra locking & re-calcing affect performance?

Reply:
This was already occurring explicitly inside the hthor/roxie/thor code (it will now only set one value instead of 3 previously)

Reply:
E.g. see deleted code in thgraphmanager.cpp